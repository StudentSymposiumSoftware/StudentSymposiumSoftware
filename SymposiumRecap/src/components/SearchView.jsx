import "./SearchView.css"

function SearchView() {
    return (
        <div id="search-container">
            Welcome to the search page
        </div>
    )
}

export default SearchView
  