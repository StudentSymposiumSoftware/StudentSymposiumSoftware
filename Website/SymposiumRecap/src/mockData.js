/*

  Mock Data has many objects of these fields:
   - Author
   - Title
   - Category
   - Major
   - Professor
   - University
   - Abstract Number

*/


const mock_data = [{ author: 'Elijah Smith',
    professor: 'Olivia Lewis',
    school: 'UMaine Machias',
    major: 'Physics',
    title: 'Machine Learning Approaches to Financial Fraud Detection',
    category: 'Social Sciences and Humanities',
    abstractNumber: 38893888 },
  { author: 'Sophia Campbell',
    professor: 'Mia Lee',
    school: 'UMaine Machias',
    major: 'Political Science',
    title: 'The Psychology of Decision-Making in Uncertain Environments',
    category: 'Social Sciences and Humanities',
    abstractNumber: 75957852 },
  { author: 'Lucy Martinez',
    professor: 'Carter Martinez',
    school: 'UMaine Machias',
    major: 'Business Administration',
    title: 'The Sociology of Online Communities in the Digital Age',
    category: 'Natural Sciences',
    abstractNumber: 48203955 },
  { author: 'Zoe Jackson',
    professor: 'Layla Perez',
    school: 'UMaine Machias',
    major: 'Psychology',
    title: 'Advancements in Human-Robot Interaction for Assistive Technologies',
    category: 'Physical and Mathematical Sciences',
    abstractNumber: 64568884 },
  { author: 'Noah King',
    professor: 'William Smith',
    school: 'UMaine Machias',
    major: 'Political Science',
    title: 'Blockchain Technology in Supply Chain Transparency',
    category: 'Biomedical Sciences',
    abstractNumber: 40679302 },
  { author: 'Joseph Baker',
    professor: 'Aria Thompson',
    school: 'UMaine Orono',
    major: 'Mathematics',
    title: 'Advancements in Human-Robot Interaction for Assistive Technologies',
    category: 'Interdisciplinary Research (research across 3 or more different departments or colleges)',
    abstractNumber: 37625940 },
  { author: 'Ava Rodriguez',
    professor: 'Victoria Thomas',
    school: 'UMaine Machias',
    major: 'Physics',
    title: 'Climate Change and Its Effects on Global Food Security',
    category: 'Physical and Mathematical Sciences',
    abstractNumber: 35716422 },
  { author: 'Henry Adams',
    professor: 'Lucy Miller',
    school: 'UMaine Machias',
    major: 'Psychology',
    title: 'The Psychology of Decision-Making in Uncertain Environments',
    category: 'Business',
    abstractNumber: 82413071 },
  { author: 'Mason Hernandez',
    professor: 'Aiden Perez',
    school: 'UMaine Orono',
    major: 'Business Administration',
    title: 'Machine Learning Approaches to Financial Fraud Detection',
    category: 'Arts',
    abstractNumber: 27882383 },
  { author: 'Emma Taylor',
    professor: 'Ethan Hernandez',
    school: 'UMaine Orono',
    major: 'Biology',
    title: 'Machine Learning Approaches to Financial Fraud Detection',
    category: 'Business',
    abstractNumber: 92969314 },
  { author: 'Aurora Martinez',
    professor: 'William Anderson',
    school: 'UMaine Machias',
    major: 'Computer Science',
    title: 'The Role of Citizen Science in Biodiversity Conservation',
    category: 'Business',
    abstractNumber: 29718536 },
  { author: 'Lucas Robinson',
    professor: 'Scarlett King',
    school: 'UMaine Orono',
    major: 'Psychology',
    title: 'The Role of Citizen Science in Biodiversity Conservation',
    category: 'Natural Sciences',
    abstractNumber: 1457768 },
  { author: 'Chloe Adams',
    professor: 'Liam Rodriguez',
    school: 'UMaine Orono',
    major: 'Physics',
    title: 'Blockchain Technology in Supply Chain Transparency',
    category: 'Business',
    abstractNumber: 14148682 },
  { author: 'Layla Young',
    professor: 'Alexander Roberts',
    school: 'UMaine Orono',
    major: 'Biology',
    title: 'Exploring the Feasibility of Space-Based Solar Power',
    category: 'Engineering and Information Sciences (including Biomedical Engineering)',
    abstractNumber: 60454722 },
  { author: 'Alexander Hall',
    professor: 'Aria Clark',
    school: 'UMaine Machias',
    major: 'Business Administration',
    title: 'Social Media Influence on Adolescent Mental Health',
    category: 'Education',
    abstractNumber: 41498009 },
  { author: 'Zoe Young',
    professor: 'Amelia Jackson',
    school: 'UMaine Orono',
    major: 'Physics',
    title: 'Analyzing the Impact of Microplastics on Marine Ecosystems',
    category: 'Engineering and Information Sciences (including Biomedical Engineering)',
    abstractNumber: 23333306 },
  { author: 'Mason Clark',
    professor: 'Scarlett Hernandez',
    school: 'UMaine Machias',
    major: 'Nursing',
    title: 'Cultural Impacts on Language Evolution',
    category: 'Physical and Mathematical Sciences',
    abstractNumber: 97656877 },
  { author: 'Henry Hall',
    professor: 'Ella Garcia',
    school: 'UMaine Machias',
    major: 'Mechanical Engineering',
    title: 'Social Media Influence on Adolescent Mental Health',
    category: 'Interdisciplinary Research (research across 3 or more different departments or colleges)',
    abstractNumber: 91417686 },
  { author: 'Samuel Scott',
    professor: 'Evelyn Green',
    school: 'UMaine Machias',
    major: 'Mathematics',
    title: 'Advancements in Human-Robot Interaction for Assistive Technologies',
    category: 'Allied Health',
    abstractNumber: 4887031 },
  { author: 'Grace Nelson',
    professor: 'Aria Scott',
    school: 'UMaine Machias',
    major: 'Nursing',
    title: 'Advancements in Human-Robot Interaction for Assistive Technologies',
    category: 'Physical and Mathematical Sciences',
    abstractNumber: 36475226 },
  { author: 'Scarlett Hernandez',
    professor: 'Matthew Wright',
    school: 'UMaine Machias',
    major: 'Mechanical Engineering',
    title: 'Machine Learning Approaches to Financial Fraud Detection',
    category: 'Arts',
    abstractNumber: 29014475 },
  { author: 'Maya Hernandez',
    professor: 'William Green',
    school: 'UMaine Machias',
    major: 'Physics',
    title: 'The Future of CRISPR in Gene Therapy',
    category: 'Physical and Mathematical Sciences',
    abstractNumber: 44667664 },
  { author: 'Elijah Rivera',
    professor: 'Harper Martinez',
    school: 'UMaine Orono',
    major: 'Computer Science',
    title: 'Analyzing the Impact of Microplastics on Marine Ecosystems',
    category: 'Social Sciences and Humanities',
    abstractNumber: 21709418 },
  { author: 'Liam Taylor',
    professor: 'Ella Walker',
    school: 'UMaine Machias',
    major: 'Political Science',
    title: 'Social Media Influence on Adolescent Mental Health',
    category: 'Physical and Mathematical Sciences',
    abstractNumber: 45822327 },
  { author: 'Mia Robinson',
    professor: 'Noah Green',
    school: 'UMaine Orono',
    major: 'Mathematics',
    title: 'The Impact of Remote Work on Organizational Productivity',
    category: 'Physical and Mathematical Sciences',
    abstractNumber: 75710656 },
  { author: 'Evelyn Rodriguez',
    professor: 'Lucas Moore',
    school: 'UMaine Orono',
    major: 'Physics',
    title: 'Deep Learning for Medical Image Analysis',
    category: 'Allied Health',
    abstractNumber: 39877467 },
  { author: 'Zoe Brown',
    professor: 'Scarlett Thompson',
    school: 'UMaine Orono',
    major: 'Physics',
    title: 'Cultural Impacts on Language Evolution',
    category: 'Biomedical Sciences',
    abstractNumber: 74596190 },
  { author: 'Sophie Thomas',
    professor: 'David Rodriguez',
    school: 'UMaine Machias',
    major: 'Mechanical Engineering',
    title: 'Advancements in Human-Robot Interaction for Assistive Technologies',
    category: 'Physical and Mathematical Sciences',
    abstractNumber: 30556122 },
  { author: 'Maya Mitchell',
    professor: 'Aiden White',
    school: 'UMaine Orono',
    major: 'Nursing',
    title: 'Advancements in Human-Robot Interaction for Assistive Technologies',
    category: 'Arts',
    abstractNumber: 75180428 },
  { author: 'Jack Anderson',
    professor: 'Harper Rodriguez',
    school: 'UMaine Orono',
    major: 'Physics',
    title: 'Analyzing the Impact of Microplastics on Marine Ecosystems',
    category: 'Natural Sciences',
    abstractNumber: 82398661 }]

export const getMockData = () => {
    return mock_data;
}