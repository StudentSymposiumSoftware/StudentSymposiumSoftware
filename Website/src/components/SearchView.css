select {
    width: 200px;
    padding: 10px;
    margin: 10px;
}

option {
    padding: 10px;
}