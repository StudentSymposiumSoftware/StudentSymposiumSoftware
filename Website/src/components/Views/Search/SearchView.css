select {
    width: 200px;
    padding: 10px;
    margin: 10px;
}

option {
    padding: 10px;
}

.search-bar{
    width: 70vw;
    border: solid white 1px;
    border-radius: 20px;
    text-align: center;
    font-size: 1.5vw;
    margin-top: 5vh;
    margin-bottom: 5vh;
    
}

.search-results{
    display: flex;
    flex-direction: column;
    gap: 1vh;
    justify-content: center;
    align-items: center;
}

.search-item{
    padding: 5px;
    width: 50vw;
    height: 10vh;
    background-color: rgba(117, 117, 117, 0.226);
    cursor: pointer;
    gap: 0.5vw;
    display: flex;
    justify-content: center;
    align-items: center;
    border-radius: 20px;
    transition: all 0.25s ease;
    border-radius: solid 1px black;
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
}

.search-item h4{
    font-size: 0.9vw;
}

.search-item h3{
    font-size: 0.8vw;
}

.search-item:hover{
    transform: scale(1.02);
    box-shadow: 0 10px 16px 0 rgba(0, 0, 0, 0.5), 0 6px 20px 0 rgba(0, 0, 0, 0.5);
}

.no-results-found{
    font-size: 2vw;
}

