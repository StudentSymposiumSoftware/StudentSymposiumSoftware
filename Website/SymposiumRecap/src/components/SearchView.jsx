import "./SearchView.css"

function SearchView(props) {

    console.log(props.data) // <-- Here's the data for your ticket

    return (
        <div id="search-container">
            Welcome to the search page
        </div>
    )
}

export default SearchView
  