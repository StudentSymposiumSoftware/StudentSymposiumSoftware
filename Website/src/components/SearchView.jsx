import "./SearchView.css";
import React, { useState } from "react";

function SearchView(props) {
    console.log(props.data); // <-- Here's the data for your ticket

    const [filters, setFilters] = useState({
        major: null,
        professor: null,
        category: null,
        university: null,
    });

    const handleChange = (e) => {
        const { name, value } = e.target;
        setFilters((prevFilters) => {
            const updatedFilters = { ...prevFilters };
            updatedFilters[name] = value;
            return updatedFilters;
        });
    };


    const getUniqueValues = (key) => {
        return [...new Set(props.data.map(item => item[key]))];
    };

    const majorOptions = getUniqueValues("major");
    const professorOptions = getUniqueValues("professor");
    const categoryOptions = getUniqueValues("category");
    const universityOptions = getUniqueValues("school");

    return (
        <div id="search-container">
            <h2>Welcome to the search page</h2>

            <div>
                <label htmlFor="major">Major:</label>
                <select
                    id="major"
                    name="major"
                    value={filters.major || ""}
                    onChange={handleChange}
                >
                    <option value="">Select a major</option>
                    {majorOptions.map((major, index) => (
                        <option key={index} value={major}>
                            {major}
                        </option>
                    ))}
                </select>
            </div>

            <div>
                <label htmlFor="professor">Professor:</label>
                <select
                    id="professor"
                    name="professor"
                    value={filters.professor || ""}
                    onChange={handleChange}
                >
                    <option value="">Select a professor</option>
                    {professorOptions.map((professor, index) => (
                        <option key={index} value={professor}>
                            {professor}
                        </option>
                    ))}
                </select>
            </div>

            <div>
                <label htmlFor="category">Category:</label>
                <select
                    id="category"
                    name="category"
                    value={filters.category || ""}
                    onChange={handleChange}
                >
                    <option value="">Select a category</option>
                    {categoryOptions.map((category, index) => (
                        <option key={index} value={category}>
                            {category}
                        </option>
                    ))}
                </select>
            </div>

            <div>
                <label htmlFor="university">University:</label>
                <select
                    id="university"
                    name="university"
                    value={filters.university || ""}
                    onChange={handleChange}
                >
                    <option value="">Select a university</option>
                    {universityOptions.map((university, index) => (
                        <option key={index} value={university}>
                            {university}
                        </option>
                    ))}
                </select>
            </div>

            <pre>{JSON.stringify(filters, null, 2)}</pre>
        </div>
    );
}

export default SearchView;
