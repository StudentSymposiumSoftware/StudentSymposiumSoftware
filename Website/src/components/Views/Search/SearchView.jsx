import "./SearchView.css";
import React, { useState, useMemo } from "react";
import { useNavigate } from "react-router-dom"

function SearchView(props) {
    const navigate = useNavigate();
    const [query, setQuery] = useState("");

    const filteredData = useMemo(() => {
        return props.data.filter(item =>
            Object.values(item).some(value =>
                String(value).toLowerCase().includes(query.toLowerCase())
            )
        )
    }, [query, props.data])

    return (
        <div id="search-container">
            <input
                type="text"
                placeholder="Search..."
                value={query}
                onChange={(e) => setQuery(e.target.value)}
                className="search-bar"
            />
            
            <div className="search-results">
                {filteredData.length > 0 ? (
                    filteredData.map((item, index) => (
                        <div key={index} className="search-item" onClick={() => navigate(`/author/${item.author}`)}>
                            <h4>#{item.abstractNumber}</h4> 
                            <h3>{item.title}</h3>      
                            <h4>by {item.author}</h4>
                        </div>
                    ))
                ) : (
                <div className="no-results-found">No results found</div>
                )}
            </div>
        
{/* 
            <div>
                <label htmlFor="major">Major:</label>
                <select
                    id="major"
                    name="major"
                    value={filters.major || ""}
                    onChange={handleChange}
                >
                    <option value="">Select a major</option>
                    {majorOptions.map((major, index) => (
                        <option key={index} value={major}>
                            {major}
                        </option>
                    ))}
                </select>
            </div>

            <div>
                <label htmlFor="professor">Professor:</label>
                <select
                    id="professor"
                    name="professor"
                    value={filters.professor || ""}
                    onChange={handleChange}
                >
                    <option value="">Select a professor</option>
                    {professorOptions.map((professor, index) => (
                        <option key={index} value={professor}>
                            {professor}
                        </option>
                    ))}
                </select>
            </div>

            <div>
                <label htmlFor="category">Category:</label>
                <select
                    id="category"
                    name="category"
                    value={filters.category || ""}
                    onChange={handleChange}
                >
                    <option value="">Select a category</option>
                    {categoryOptions.map((category, index) => (
                        <option key={index} value={category}>
                            {category}
                        </option>
                    ))}
                </select>
            </div>

            <div>
                <label htmlFor="university">University:</label>
                <select
                    id="university"
                    name="university"
                    value={filters.university || ""}
                    onChange={handleChange}
                >
                    <option value="">Select a university</option>
                    {universityOptions.map((university, index) => (
                        <option key={index} value={university}>
                            {university}
                        </option>
                    ))}
                </select>
            </div> */}

       
        </div>
    );
}

export default SearchView;
