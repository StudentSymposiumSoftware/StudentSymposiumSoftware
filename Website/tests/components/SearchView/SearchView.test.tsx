import React from 'react';
import { render } from "@testing-library/react";
import { expect, it } from "vitest";

import SearchView from '../../../src/components/SearchView';
import mockData from '../../../src/mockData'

it('SearchView renders correctly', () => {
    const tree = render(<SearchView data={mockData}/>)
    expect(tree).toMatchSnapshot();
});