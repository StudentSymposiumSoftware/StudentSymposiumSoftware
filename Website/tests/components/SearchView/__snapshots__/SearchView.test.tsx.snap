// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SearchView renders correctly 1`] = `
<div>
  <div
    id="search-container"
  >
    <input
      class="search-bar"
      placeholder="Search..."
      type="text"
      value=""
    />
    <div
      class="search-results"
    >
      <div
        class="search-item"
      >
        <h4>
          #
          38893888
        </h4>
        <h3>
          Machine Learning Approaches to Financial Fraud Detection
        </h3>
        <h4>
          by 
          Elijah Smith
        </h4>
      </div>
      <div
        class="search-item"
      >
        <h4>
          #
          75957852
        </h4>
        <h3>
          The Psychology of Decision-Making in Uncertain Environments
        </h3>
        <h4>
          by 
          Sophia Campbell
        </h4>
      </div>
      <div
        class="search-item"
      >
        <h4>
          #
          48203955
        </h4>
        <h3>
          The Sociology of Online Communities in the Digital Age
        </h3>
        <h4>
          by 
          Lucy Martinez
        </h4>
      </div>
      <div
        class="search-item"
      >
        <h4>
          #
          64568884
        </h4>
        <h3>
          Advancements in Human-Robot Interaction for Assistive Technologies
        </h3>
        <h4>
          by 
          Zoe Jackson
        </h4>
      </div>
      <div
        class="search-item"
      >
        <h4>
          #
          40679302
        </h4>
        <h3>
          Blockchain Technology in Supply Chain Transparency
        </h3>
        <h4>
          by 
          Noah King
        </h4>
      </div>
      <div
        class="search-item"
      >
        <h4>
          #
          37625940
        </h4>
        <h3>
          Advancements in Human-Robot Interaction for Assistive Technologies
        </h3>
        <h4>
          by 
          Joseph Baker
        </h4>
      </div>
      <div
        class="search-item"
      >
        <h4>
          #
          35716422
        </h4>
        <h3>
          Climate Change and Its Effects on Global Food Security
        </h3>
        <h4>
          by 
          Ava Rodriguez
        </h4>
      </div>
      <div
        class="search-item"
      >
        <h4>
          #
          82413071
        </h4>
        <h3>
          The Psychology of Decision-Making in Uncertain Environments
        </h3>
        <h4>
          by 
          Henry Adams
        </h4>
      </div>
      <div
        class="search-item"
      >
        <h4>
          #
          27882383
        </h4>
        <h3>
          Machine Learning Approaches to Financial Fraud Detection
        </h3>
        <h4>
          by 
          Mason Hernandez
        </h4>
      </div>
      <div
        class="search-item"
      >
        <h4>
          #
          92969314
        </h4>
        <h3>
          Machine Learning Approaches to Financial Fraud Detection
        </h3>
        <h4>
          by 
          Emma Taylor
        </h4>
      </div>
      <div
        class="search-item"
      >
        <h4>
          #
          29718536
        </h4>
        <h3>
          The Role of Citizen Science in Biodiversity Conservation
        </h3>
        <h4>
          by 
          Aurora Martinez
        </h4>
      </div>
      <div
        class="search-item"
      >
        <h4>
          #
          1457768
        </h4>
        <h3>
          The Role of Citizen Science in Biodiversity Conservation
        </h3>
        <h4>
          by 
          Lucas Robinson
        </h4>
      </div>
      <div
        class="search-item"
      >
        <h4>
          #
          14148682
        </h4>
        <h3>
          Blockchain Technology in Supply Chain Transparency
        </h3>
        <h4>
          by 
          Chloe Adams
        </h4>
      </div>
      <div
        class="search-item"
      >
        <h4>
          #
          60454722
        </h4>
        <h3>
          Exploring the Feasibility of Space-Based Solar Power
        </h3>
        <h4>
          by 
          Layla Young
        </h4>
      </div>
      <div
        class="search-item"
      >
        <h4>
          #
          41498009
        </h4>
        <h3>
          Social Media Influence on Adolescent Mental Health
        </h3>
        <h4>
          by 
          Alexander Hall
        </h4>
      </div>
      <div
        class="search-item"
      >
        <h4>
          #
          23333306
        </h4>
        <h3>
          Analyzing the Impact of Microplastics on Marine Ecosystems
        </h3>
        <h4>
          by 
          Zoe Young
        </h4>
      </div>
      <div
        class="search-item"
      >
        <h4>
          #
          97656877
        </h4>
        <h3>
          Cultural Impacts on Language Evolution
        </h3>
        <h4>
          by 
          Mason Clark
        </h4>
      </div>
      <div
        class="search-item"
      >
        <h4>
          #
          91417686
        </h4>
        <h3>
          Social Media Influence on Adolescent Mental Health
        </h3>
        <h4>
          by 
          Henry Hall
        </h4>
      </div>
      <div
        class="search-item"
      >
        <h4>
          #
          4887031
        </h4>
        <h3>
          Advancements in Human-Robot Interaction for Assistive Technologies
        </h3>
        <h4>
          by 
          Samuel Scott
        </h4>
      </div>
      <div
        class="search-item"
      >
        <h4>
          #
          36475226
        </h4>
        <h3>
          Advancements in Human-Robot Interaction for Assistive Technologies
        </h3>
        <h4>
          by 
          Grace Nelson
        </h4>
      </div>
      <div
        class="search-item"
      >
        <h4>
          #
          29014475
        </h4>
        <h3>
          Machine Learning Approaches to Financial Fraud Detection
        </h3>
        <h4>
          by 
          Scarlett Hernandez
        </h4>
      </div>
      <div
        class="search-item"
      >
        <h4>
          #
          44667664
        </h4>
        <h3>
          The Future of CRISPR in Gene Therapy
        </h3>
        <h4>
          by 
          Maya Hernandez
        </h4>
      </div>
      <div
        class="search-item"
      >
        <h4>
          #
          21709418
        </h4>
        <h3>
          Analyzing the Impact of Microplastics on Marine Ecosystems
        </h3>
        <h4>
          by 
          Elijah Rivera
        </h4>
      </div>
      <div
        class="search-item"
      >
        <h4>
          #
          45822327
        </h4>
        <h3>
          Social Media Influence on Adolescent Mental Health
        </h3>
        <h4>
          by 
          Liam Taylor
        </h4>
      </div>
      <div
        class="search-item"
      >
        <h4>
          #
          75710656
        </h4>
        <h3>
          The Impact of Remote Work on Organizational Productivity
        </h3>
        <h4>
          by 
          Mia Robinson
        </h4>
      </div>
      <div
        class="search-item"
      >
        <h4>
          #
          39877467
        </h4>
        <h3>
          Deep Learning for Medical Image Analysis
        </h3>
        <h4>
          by 
          Evelyn Rodriguez
        </h4>
      </div>
      <div
        class="search-item"
      >
        <h4>
          #
          74596190
        </h4>
        <h3>
          Cultural Impacts on Language Evolution
        </h3>
        <h4>
          by 
          Zoe Brown
        </h4>
      </div>
      <div
        class="search-item"
      >
        <h4>
          #
          30556122
        </h4>
        <h3>
          Advancements in Human-Robot Interaction for Assistive Technologies
        </h3>
        <h4>
          by 
          Sophie Thomas
        </h4>
      </div>
      <div
        class="search-item"
      >
        <h4>
          #
          75180428
        </h4>
        <h3>
          Advancements in Human-Robot Interaction for Assistive Technologies
        </h3>
        <h4>
          by 
          Maya Mitchell
        </h4>
      </div>
      <div
        class="search-item"
      >
        <h4>
          #
          82398661
        </h4>
        <h3>
          Analyzing the Impact of Microplastics on Marine Ecosystems
        </h3>
        <h4>
          by 
          Jack Anderson
        </h4>
      </div>
      <div
        class="search-item"
      >
        <h4>
          #
          82398661
        </h4>
        <h3>
          Analyzing the Impact of Microplastics on Marine Ecosystems
        </h3>
        <h4>
          by 
          Harper Rodriguez
        </h4>
      </div>
    </div>
  </div>
</div>
`;

exports[`Searching should properly search abstracts 1`] = `
<div>
  <div
    id="search-container"
  >
    <input
      class="search-bar"
      placeholder="Search..."
      type="text"
      value="Fraud"
    />
    <div
      class="search-results"
    >
      <div
        class="search-item"
      >
        <h4>
          #
          38893888
        </h4>
        <h3>
          Machine Learning Approaches to Financial Fraud Detection
        </h3>
        <h4>
          by 
          Elijah Smith
        </h4>
      </div>
      <div
        class="search-item"
      >
        <h4>
          #
          27882383
        </h4>
        <h3>
          Machine Learning Approaches to Financial Fraud Detection
        </h3>
        <h4>
          by 
          Mason Hernandez
        </h4>
      </div>
      <div
        class="search-item"
      >
        <h4>
          #
          92969314
        </h4>
        <h3>
          Machine Learning Approaches to Financial Fraud Detection
        </h3>
        <h4>
          by 
          Emma Taylor
        </h4>
      </div>
      <div
        class="search-item"
      >
        <h4>
          #
          29014475
        </h4>
        <h3>
          Machine Learning Approaches to Financial Fraud Detection
        </h3>
        <h4>
          by 
          Scarlett Hernandez
        </h4>
      </div>
    </div>
  </div>
</div>
`;
